kurtosis-module/impl/ethereum_genesis_generator/genesis_generator.go

```diff
@@ -0,0 +1,218 @@
+	expectedCLMnemonicsYmlFilepath = expectedCLConfigDirpathOnService + "/mnemonics.yaml"
+
+	// The generator container hardcodes the location where it'll write the output to; this is the location
+	outputGenesisDataDirpathOnGeneratorContainer = "/data"
+
+	// Location, relative to the root of shared dir, where the genesis data will be copied to from the generator container
+	//  so that the module container can store and use it
+	genesisDataRelDirpathInSharedDir = "data"
+
+	// This is the entrypoint that the Dockerfile uses (though we override it so that we can do some extra work
+	//  before it runs)
+	entrypointFromDockerfile = "/work/entrypoint.sh"
+
+	generationCommandExpectedExitCode = 0
+
+	// Paths, *relative to the root of the output genesis data directory, where the generator writes data
+	outputGethGenesisConfigRelFilepath = "el/geth.json"
+	outputClGenesisConfigRelDirpath = "cl"
+
+	containerStopTimeoutSeconds = 3
+)
+// We run the genesis generation as an exec command instead, so that we get immediate feedback if it fails
+var entrypoingArgs = []string{
+	"sleep",
+	"99999",
+}
+
+type elGenesisConfigTemplateData struct {
+	NetworkId string
+}
+type clGenesisConfigTemplateData struct {
+	NetworkId string
+}
+
+func GenerateELAndCLGenesisConfig(
+	enclaveCtx *enclaves.EnclaveContext,
+	elGenesisConfigYmlTemplate *template.Template,
+	clGenesisConfigYmlTemplate *template.Template,
+	clMnemonicsYmlFilepathOnModuleContainer string,
+	networkId string,
+) (
+	resultGethELGenesisJSONFilepath string,
+	resultCLGenesisDataDirpath string,
+	resultErr error,
+) {
+	serviceCtx, err := enclaveCtx.AddService(serviceId, getContainerConfig)
+	if err != nil {
+		return "", "", stacktrace.Propagate(err, "An error occurred launching the Ethereum genesis-generating container with service ID '%v'", serviceId)
+	}
+
+	gethGenesisJsonFilepath, clGenesisDataDirpath, err := generateGenesisData(serviceCtx, networkId, elGenesisConfigYmlTemplate, clGenesisConfigYmlTemplate, clMnemonicsYmlFilepathOnModuleContainer)
+	if err != nil {
+		return "", "", stacktrace.Propagate(err, "An error occurred generating genesis data")
+	}
+
+	if err := enclaveCtx.RemoveService(serviceId, containerStopTimeoutSeconds); err != nil {
+		logrus.Errorf(
+			"An error occurred stopping the genesis generation service with ID '%v' and timeout '%vs'; you'll need to stop it manually",
+			serviceId,
+			containerStopTimeoutSeconds,
+		)
+	}
+
+	return gethGenesisJsonFilepath, clGenesisDataDirpath, nil
+}
+
+func generateGenesisData(
+	serviceCtx *services.ServiceContext,
+	networkId string,
+	gethGenesisConfigYmlTemplate *template.Template,
+	clGenesisConfigYmlTemplate *template.Template,
+	clMnemonicsYmlFilepathOnModuleContainer string,
+) (
+	resultGethGenesisJsonFilepathOnModuleContainer string,
+	resultClConfigDataDirpathOnModuleContainer string,
+	resultErr error,
+) {
+	elTemplateData := elGenesisConfigTemplateData{
+		NetworkId: networkId,
```

Comment:
I think that we also have to add the "terminal_total_difficulty" property, so it can be dynamic
![image](https://user-images.githubusercontent.com/29951623/146377194-2d9ae12d-9b63-4f27-adbe-12f157d48538.png)

Reply:
Good spot; already done in the `mieubrisse/teku` branch I'm working on!